test/v1alpha1/service.go

```diff
@@ -90,12 +119,32 @@ func GetResourceObjects(clients *test.Clients, names test.ResourceNames) (*Resou
 
 // CreateRunLatestServiceReady creates a new Service in state 'Ready'. This function expects Service and Image name passed in through 'names'.
 // Names is updated with the Route and Configuration created by the Service and ResourceObjects is returned with the Service, Route, and Configuration objects.
+// If this function is called with https == true, the gateway MUST be restored afterwards.
 // Returns error if the service does not come up correctly.
-func CreateRunLatestServiceReady(t *testing.T, clients *test.Clients, names *test.ResourceNames, fopt ...rtesting.ServiceOption) (*ResourceObjects, error) {
+func CreateRunLatestServiceReady(t *testing.T, clients *test.Clients, names *test.ResourceNames, https bool, fopt ...rtesting.ServiceOption) (*ResourceObjects, error) {
 	if names.Image == "" {
 		return nil, fmt.Errorf("expected non-empty Image name; got Image=%v", names.Image)
 	}
 
+	if https {
+		tlsOptions := &v1alpha3.TLSOptions{
+			Mode:              v1alpha3.TLSModeSimple,
+			PrivateKey:        "/etc/istio/ingressgateway-certs/tls.key",
+			ServerCertificate: "/etc/istio/ingressgateway-certs/tls.crt",
+		}
+		servers := []v1alpha3.Server{{
+			Hosts: []string{"*"},
+			Port: v1alpha3.Port{
+				Name:     "standard-https",
+				Number:   443,
+				Protocol: v1alpha3.ProtocolHTTPS,
+			},
+			TLS: tlsOptions,
+		}}
+		setupHTTPS(t, clients.KubeClient, []string{names.Service + "." + GetDomain(t, clients)})
```

Comment:
Can we use names.URL (https://github.com/knative/serving/blob/fdcec3ba90b4e875b2b171be35393db22a14bca2/test/crd.go#L36) here?

Reply:
Hmm names.URL actually won't be set until after the service is created, so I think this is the best way.

Reply:
I am not sure if we can move the part of configuring https (https://github.com/knative/serving/pull/5157/files#diff-cd1624ee494bbca6fd2e1e1a908d0dd3R131) after `validateCreatedServiceStatus` function (https://github.com/knative/serving/pull/5157/files#diff-cd1624ee494bbca6fd2e1e1a908d0dd3R174).
If we can, then `names.URL` will be available after `validateCreatedServiceStatus` function. And then we can use `names.URL` instead of using `GetDomain`.

Reply:
@ZhiminXiang yes that does work. Thank you for the suggestion!